### PR TITLE
Initial commit to bring in the docs

### PR DESCRIPTION
**Pull request by HarelM**
This is related to #2150.
It only adds the relevant parts to this repo as a commit that won't affect anything at that point.
After this I'll create a branch the will allow to slowly convert the tsdocs.
Once all the docs are converted we will be able to merge this branch and create a new documentation site.

Things that were added as part of this PR:
1. mkdocs config file
2. docs generation script
3. images generation script
4. All the examples are now part of this repo for both documentation and debugging.
5. some lint fixes

How to review this:
You can ignore the html and png files as they were copied as is from the docs repo.
Other files (~10) are the interesting stuff.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
+# Introduction
+
+MapLibre GL is a TypeScript library that uses WebGL2 to render interactive maps from vector tiles in a browser. The customization of the map comply with the [MapLibre Style Spec](https://maplibre.org/maplibre-style-spec). It is part of the [MapLibre ecosystem](https://github.com/maplibre), with a pendant for Mobile, Desktop, Servers called [MapLibre Native](https://maplibre.org/projects/maplibre-native/).
```
Comment (birkskyum) [inline docs/README.md:3]: GL -> GL JS

File: mkdocs.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+site_name: MapLibre GL
```
Comment (birkskyum) [inline mkdocs.yml:1]: GL -> GL JS

File: typedoc.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    "out": "./docs/API",
+    "excludeExternals": true,
+    "excludeNotDocumented": true,
+    "internalModule": "maplibregl",
+    "entryPoints": ["./src/index.ts"],
+    "navigation": {
+        "includeCategories": false,
+        "includeGroups": true
+    },
+    "name": "MapLibre GL",
```
Comment (birkskyum) [inline typedoc.json:17]: GL -> GL JS

File: typedoc.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    "githubPages": false,
```
Comment (birkskyum) [inline typedoc.json:18]: Should this maybe be true?

File: build/generate-doc-images.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
+import path from 'path';
+import fs from 'fs';
+import puppeteer from 'puppeteer';
+import packageJson from '../package.json' assert { type: 'json' };
+
+const exampleName = process.argv[2];
+const examplePath = path.resolve('test', 'examples');
+
+const browser = await puppeteer.launch({headless: exampleName === 'all'});
```
[the remark this answers is not in the document]
Comment (birkskyum) [reply build/generate-doc-images.ts:9]: How come headless: 'new' isn't used in all cases?

[the remark this answers is not in the document]
Comment (HarelM) [reply build/generate-doc-images.ts:9]: No clue, I mainly copied the script over...

File: typedoc.json (hunk 4 above)
Comment (HarelM) [reply typedoc.json:18]: This removes a file that is not needed as the docs will be converted to html by mkdocs later on the process.

**Review by birkskyum — APPROVED** [review]

**Pull request merged**